mq/inmemory/mq.h

```diff
@@ -101,6 +101,13 @@ struct ExportMessageImpl<T_CONSUMER, T_MESSAGE, false> {
 
 template <typename T_CONSUMER, typename T_MESSAGE>
 void ExportMessage(T_CONSUMER& consumer, T_MESSAGE&& message, size_t dropped_count) {
+  // Note that the allowed syntax for `OnMessage()` is:
+  //   either `OnMessage(const T_MESSAGE& message [, size_t dropped_count]);`
+  //   or     `OnMessage(T_MESSAGE&& message [, size_t dropped_count]);`
```

Comment:
It still calls the constructor. Extension :)

```c++
#include <iostream>

struct A {
  int x;
  A(const int x) : x(x) {
    std::cout << "Create A. x = " << x << std::endl;
  }

  A(const A& a) : x(a.x) {
    std::cout << "Copy" << std::endl;
  }

  A(A&& a) : x(std::move(a.x)) {
    std::cout << "Move" << std::endl;
  }
};

void foo(const A& a) {
  std::cout << "x = " << a.x << std::endl;
}

void bar(A a) {
  std::cout << "x = " << a.x << std::endl;
}

void baz(A&& a) {
  std::cout << "x = " << a.x << std::endl;
}

int main() {
  A a1(1);
  foo(a1);
  bar(a1);
  // Constructor called.
  bar(std::move(a1));
  // Constructor not called.
  baz(std::move(a1));
}
```
Create A. x = 1
x = 1
Copy
x = 1
Move
x = 1
x = 1
```

```